doc/source/testing.rst

```diff
@@ -1,51 +1,124 @@
+``minikube``
+^^^^^^^^^^^^
+1. Install ``minikube`` by following online instructions, or use::
 
-    it most likely means you have docker running on your host using the same
-    IP range minikube is using. You can fix this by editing your
-    ``/etc/docker/daemon.json`` file to add the following:
+      make kubectl-bootstrap
+2. Start ``minikube`` with::
 
-    .. code-block:: json
+      minikube start
+3. (recommended) to run containerized tests in a pod, make sure you have ``docker`` installed and run ``eval $(shell minikube docker-env)`` before building the Docker image
+4. (optional) if instead you want to run tests locally, you don't need ``docker``, but you will need to make it possible for your host to be able to talk to the pods on minikube - see section below.
+5. Create a namespace and role bindings for testing::
 
-       {
-           "bip": "172.19.1.1/16"
-       }
+      make k8s-deploy
 
-    If some JSON already exists in that file, make sure to just add the
-    ``bip`` key rather than replace it all. The final file needs to be valid
-    JSON.
+(Optional) Configure network access to the ``minikube`` pods
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+.. note::  This step may vary depending on your network configuration, and it may be easier to run containerized tests in-cluster instead
 
-    Once edited, restart docker with ``sudo systemctl restart docker``. It
-    should come up using a different IP range, and you can run the
-    ``sudo ip route add`` command again. Note that restarting docker will
-    restart all your running containers by default.
+- On Linux::
 
-3.  Run tests::
+   sudo ip route add 172.17.0.0/16 via $(minikube ip)
 
-       py.test dask_kubernetes --worker-image <worker-image>
+- On OS X::
 
-    where:
+   sudo route -n add -net 172.17.0.0/16 $(minikube ip)
 
-      - ``<worker-image>`` is a docker image that has the same version of
-        python and libraries installed as your host
-      - ``<cluster-name>`` is a identifying name that you give to this
-        specific cluster
-      - ``<namespace>`` is the namespace you created in step 3
+If you get an error message like the following::
+
+   RTNETLINK answers: File exists
+
+it most likely means you have docker running on your host using the same
+IP range minikube is using. You can fix this by editing your
+``/etc/docker/daemon.json`` file to add the following:
+
+.. code-block:: json
+
+   {
+      "bip": "172.19.1.1/16"
+   }
+
+If some JSON already exists in that file, make sure to just add the
+``bip`` key rather than replace it all. The final file needs to be valid
+JSON.
+
+Once edited, restart docker with ``sudo systemctl restart docker``. It
+should come up using a different IP range, and you can run the
+``sudo ip route add`` command again. Note that restarting docker will
+restart all your running containers by default.
+
+``kind``
+^^^^^^^^
+1. Install ``kind`` by following online instructions, or use::
```

Comment:
Can you link to kind install instructions.

Reply:
Also note that kind requires docker.

Reply:
:+1: :+1: